docs/src/prototypes/mentions/MentionsEditor.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from 'react'
+import * as _ from 'lodash'
+import keyboardKey from 'keyboard-key'
+
+import { atMentionItems, AtMentionItem } from './dataMocks'
+import { insertTextAtCursorPosition } from './utils'
+import { PortalAtCursorPosition } from './PortalAtCursorPosition'
+
+export interface MentionsContainerProps {
+  items?: AtMentionItem[]
+  searchQuery?: string
+  open?: boolean
+  onInputKeyDown?: (e: React.KeyboardEvent) => void
+  onOpenChange?: (e: React.SyntheticEvent, data: { open?: boolean }) => void
+  onSearchQueryChange?: (e: React.SyntheticEvent, data: { searchQuery: string }) => void
+  onSelectedChange?: (e: React.SyntheticEvent, data: { searchQuery: string }) => void
+}
+
+interface MentionsEditorState {
+  open?: boolean
+  searchQuery?: string
+}
+
+const editorStyle: React.CSSProperties = {
+  backgroundColor: '#eee',
+  borderRadius: '5px',
+  border: '1px dashed grey',
+  padding: '5px',
+  minHeight: '100px',
+  outline: 0,
+}
+
+const withMentionsEditor = <P extends MentionsContainerProps = MentionsContainerProps>(
+  MentionsContainerComponent: React.ComponentType<MentionsContainerProps>,
+) =>
+  class extends React.Component<P, MentionsEditorState> {
+    private readonly initialState: MentionsEditorState = {
+      open: false,
+      searchQuery: '',
+    }
+
+    private contendEditableRef = React.createRef<HTMLDivElement>()
```

Comment:
nit: remove modifiers and reorder methods, please

Reply:
I will, just FYI I'm just moving this code and not actually adding the modifiers (they were already there)

Reply:
done, thanks